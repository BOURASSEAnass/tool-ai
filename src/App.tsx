import { lazy, Suspense } from "react";
import { Route, Routes } from "react-router-dom";
import Container from "./components/Container/Container";
import Footer from "./components/Footer/Footer";
import Header from "./components/Header/Header";
import ToolLayout from "./layout/ToolLayout/ToolLayout";
import QuestionsPage from "./pages/QuestionsPage/QuestionsPage";
import SpamCheckerPage from "./pages/SpamCheckerPage/SpamCheckerPage";

const HomePage = lazy(() => import("./pages/HomePage/HomePage"));

function App() {
  return (
    <div className="bg-background min-h-screen flex flex-col justify-between min-h py-8 gap-8">
      <Header />
      <Routes>
        <Route
          path="/"
          element={
            <Suspense fallback={<div>Loading...</div>}>
              <HomePage />
            </Suspense>
          }
        />
        <Route
          path="/tools/spam-checker"
          element={
            <ToolLayout>
              <Container>
                <SpamCheckerPage />
              </Container>
            </ToolLayout>
          }
        />
        <Route
          path="/tools/language-questions"
          element={
            <ToolLayout>
              <Container>
                <QuestionsPage />
              </Container>
            </ToolLayout>
          }
        />
        <Route path="*" element={<div>404 Not Found</div>} />
      </Routes>

      <Footer />
    </div>
  );
}

export default App;
