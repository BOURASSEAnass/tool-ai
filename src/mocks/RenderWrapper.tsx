import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "../redux/store";

interface RenderWrapperProps {
  children: JSX.Element | JSX.Element[];
}

const RenderWrapper = ({ children }: RenderWrapperProps) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};

export default RenderWrapper;
