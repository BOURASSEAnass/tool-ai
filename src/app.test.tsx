import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import path from "path";
import App from "./App";
import RenderWrapper from "./mocks/RenderWrapper";

describe("Given the app", () => {
  describe("When the app is rendered", () => {
    test("Then it matches the snapshot", () => {
      const view = render(<App />, {
        wrapper: RenderWrapper,
      });
      expect(view).toMatchSnapshot();
    });
  });

  describe("When click on button 'Try' of Spam checker tool", () => {
    test("Then it should show a heading level 2 with text 'Spam checker'", async () => {
      const expectedText = "Spam Checker";
      const expectedPath = "/tools/spam-checker";
      const nameLinkTool = "Try it";

      render(<App />, {
        wrapper: RenderWrapper,
      });

      const tryItLinks = screen.getAllByRole("link", { name: nameLinkTool });

      const linkTryIt = tryItLinks.find((link) => {
        return link.getAttribute("href") === expectedPath;
      });

      await userEvent.click(linkTryIt!);

      const heading = screen.getByRole("heading", {
        level: 2,
        name: expectedText,
      });

      expect(heading).toBeInTheDocument();
    });
  });

  describe("When click on button 'Try' of Question", () => {
    test("Then it should show a heading level 2 with text 'Question'", async () => {
      const expectedText = "Questions";
      const expectedPath = "/tools/language-questions";
      const nameLinkTool = "Questions";

      render(<App />, {
        wrapper: RenderWrapper,
      });

      const tryItLinks = screen.getAllByRole("link", { name: nameLinkTool });

      const linkTryIt = tryItLinks.find((link) => {
        return link.getAttribute("href") === expectedPath;
      });

      await userEvent.click(linkTryIt!);

      const heading = screen.getByRole("heading", {
        level: 2,
        name: expectedText,
      });

      expect(heading).toBeInTheDocument();
    });
  });
});
