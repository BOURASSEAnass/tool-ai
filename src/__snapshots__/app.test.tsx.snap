// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the app When the app is rendered Then it matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-background min-h-screen flex flex-col justify-between min-h py-8 gap-8"
      >
        <header
          class="flex justify-center"
        >
          <a
            href="/"
          >
            <h1
              class="text-6xl text-tertiary font-extrabold"
            >
              ToolAI
            </h1>
          </a>
        </header>
        <div>
          Loading...
        </div>
        <footer
          class="text-tertiary flex flex-col items-center gap-4 text-center"
        >
          <span
            class="first-letter:uppercase"
          >
            This is a submission for
             
            <a
              class="underline"
              href="https://github.com/midudev/midu-cohere-hackathon"
              rel="noreferrer"
              target="_blank"
            >
              midudev hackaton
            </a>
          </span>
          <span>
            Designed and developed by
             
            <a
              class="underline"
              href="https://github.com/Jacky16"
              rel="noreferrer"
              target="_blank"
            >
              Adrià Martínez
            </a>
             
            with ❤️
          </span>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-background min-h-screen flex flex-col justify-between min-h py-8 gap-8"
    >
      <header
        class="flex justify-center"
      >
        <a
          href="/"
        >
          <h1
            class="text-6xl text-tertiary font-extrabold"
          >
            ToolAI
          </h1>
        </a>
      </header>
      <div>
        Loading...
      </div>
      <footer
        class="text-tertiary flex flex-col items-center gap-4 text-center"
      >
        <span
          class="first-letter:uppercase"
        >
          This is a submission for
           
          <a
            class="underline"
            href="https://github.com/midudev/midu-cohere-hackathon"
            rel="noreferrer"
            target="_blank"
          >
            midudev hackaton
          </a>
        </span>
        <span>
          Designed and developed by
           
          <a
            class="underline"
            href="https://github.com/Jacky16"
            rel="noreferrer"
            target="_blank"
          >
            Adrià Martínez
          </a>
           
          with ❤️
        </span>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
